x-pack/plugins/cloud/public/plugin.tsx

```diff
@@ -270,48 +245,25 @@ export class CloudPlugin implements Plugin<CloudSetup> {
    * Set up the Analytics context providers.
    * @param analytics Core's Analytics service. The Setup contract.
    * @param security The security plugin.
-   * @param executionContextPromise Core's executionContext's start contract.
-   * @param esOrgId The Cloud Org ID.
+   * @param cloudId The Cloud Org ID.
    * @private
    */
-  private async setupTelemetryContext({
-    analytics,
-    security,
-    executionContextPromise,
-    cloudId,
-  }: SetupTelemetryContextDeps) {
-    // Some context providers can be moved to other places for better domain isolation.
-    // Let's use https://github.com/elastic/kibana/issues/125690 for that purpose.
-    analytics.registerContextProvider({
-      name: 'kibana_version',
-      context$: of({ version: this.initializerContext.env.packageInfo.version }),
-      schema: { version: { type: 'keyword', _meta: { description: 'The version of Kibana' } } },
-    });
+  private setupTelemetryContext(
+    analytics: AnalyticsServiceSetup,
+    security?: Pick<SecurityPluginSetup, 'authc'>,
+    cloudId?: string
+  ) {
+    registerCloudDeploymentIdAnalyticsContext(analytics, cloudId);
 
-    analytics.registerContextProvider({
-      name: 'cloud_org_id',
-      context$: of({ cloudId }),
-      schema: {
-        cloudId: {
-          type: 'keyword',
-          _meta: { description: 'The Cloud ID', optional: true },
-        },
-      },
-    });
-
-    // This needs to be called synchronously to be sure that we populate the user ID soon enough to make sessions merging
-    // across domains work
     if (security) {
       analytics.registerContextProvider({
         name: 'cloud_user_id',
-        context$: from(loadUserId({ getCurrentUser: security.authc.getCurrentUser })).pipe(
-          filter((userId): userId is string => Boolean(userId)),
-          exhaustMap(async (userId) => {
-            const { sha256 } = await import('js-sha256');
-            // Join the cloud org id and the user to create a truly unique user id.
-            // The hashing here is to keep it at clear as possible in our source code that we do not send literal user IDs
-            return { userId: sha256(cloudId ? `${cloudId}:${userId}` : `${userId}`) };
-          })
+        context$: from(security.authc.getCurrentUser()).pipe(
+          map((user) => user.username),
+          // Join the cloud org id and the user to create a truly unique user id.
+          // The hashing here is to keep it at clear as possible in our source code that we do not send literal user IDs
+          map((userId) => ({ userId: sha256(cloudId ? `${cloudId}:${userId}` : `${userId}`) })),
```

Comment:
We are discussing to change this piece of logic. But we'll do that in a follow-up PR.

Reply:
https://github.com/elastic/kibana/issues/131144